### DISCUSSION TITLE
BT-03-notice is not the listName of the BT-02-notice

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#537] [Q&A] [answered]
When using the SDK these fields are generated in the physical model as follows

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/ff330dd6-f6b2-47ad-be53-fe17128b3670)

And they should be generated as follows:

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/5545d895-f223-4e25-aeb9-90bacddb2397)

When validating the generated XML with the SDK, it gives the following error:

See below the full error:\n[{\"line\":51,\"column\":45,\"message\":\"cvc-complex-type.2.4.a: Invalid content was found starting with element '{\\\"urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2\\\":NoticeTypeCode}'. One of '{\\\"urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2\\\":NoticeLanguageCode, \\\"urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2\\\":AdditionalNoticeLanguage, \\\"urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2\\\":DocumentReference, \\\"urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2\\\":Signature, \\\"urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2\\\":ContractingParty}' is expected.\"},

Thanks in advance

**Comment by elerik** [comment, marked as answer]
Although it has code list with id "notice-type" and "form-type" (i believe this is how you get those elements at the end) and i agreee with you, well it is what it is. Lets take a look in to it.

"id" : "BT-02-notice",
"xpathRelative" : "cbc:NoticeTypeCode",
-> Value of Field represent value of element 

"id" : "BT-03-notice",
"xpathRelative" : "cbc:NoticeTypeCode/@listName",
-> value of field represent value of attribute. 
 /@listName represent information about that this field is pure attribute value of element NoticeTypeCode. 

Another example but different (little bit)
[cbc:ID/@schemeName='Lot']
Value of Field represent value of element but it has defined attribute schemeName with value Lot. 

Anyway watch out its not so unique. this behaviour is also for example for 
"id" : "BT-26(m)-Procedure",
"xpathRelative" : "@listName"
which value of field is "cpv" (only one avaiable option) so final xml element looks like 
<cbc:ItemClassificationCode listName="cpv">77310000</cbc:ItemClassificationCode>

but look at this for example. They are same but different in definition. Unfortunately there are many of those inconstinaces.
"id" : "BT-26(a)-Procedure",
"xpathRelative" : "cbc:ItemClassificationCode/@listName",
which value of field is "cpv" (only one avaiable option) so final xml element looks like 
<cbc:ItemClassificationCode listName="cpv">77310000</cbc:ItemClassificationCode>

There are multiple, i call it patterns, to recognize what  to do with field itself. If its value, if its value of attribute if its value but subchild has this or that attribute, and so on. Good is, its not so difficult how it looks like on first sight and its manageable with not so complex logic even there are many inconstances (so far they are manageable)
